squad-server/plugins/mapvote/components/mapvote-vote-engine.js

```diff
@@ -0,0 +1,138 @@
+import { MAP_VOTE_NEW_WINNER, MAP_VOTE_START, MAP_VOTE_END } from 'mapvote/constants';
+import EventEmitter from 'events';
+
+export default class MapvoteVoteEngine extends EventEmitter {
```

Comment:
May I suggest we use a more OOP approach no we can and move away from the `MapVote` class and event emitter?

For example, when a new map is voted for both options are going to take a layer name, cast the vote count, find the new winner, update the next map, and announce. An example of whether they differ would be in the announcement of a new winner as `mapvote-123` map need to announce the layer numbers. So we should have a method in a base class called, for example, `announceCurrentWinner` that simply announces the winner and is called at the end of the above logic. `mapvote-123` can then overwrite this method to include the layer numbers.